File: examples/python/tag_policies.py
# --8<-- [start:list_tag_policies]
from unitycatalog_client import UnityCatalogClient

client = UnityCatalogClient(base_url="http://localhost:8080")
policies = client.list_tag_policies()
for policy in policies:
    print(policy.tag_key)
# --8<-- [end:list_tag_policies]


# --8<-- [start:create_tag_policy]
def create_tag_policy_example() -> None:
    from unitycatalog_client import TagPolicy, UnityCatalogClient, Value

    client = UnityCatalogClient(base_url="http://localhost:8080")
    policy = client.create_tag_policy(
        TagPolicy(
            tag_key="classification",
            description="Data sensitivity level",
            values=[Value(name="public"), Value(name="restricted")],
        )
    )
    print(f"Created: {policy.tag_key}")


# --8<-- [end:create_tag_policy]


# --8<-- [start:get_tag_policy]
def get_tag_policy_example() -> None:
    from unitycatalog_client import UnityCatalogClient

    client = UnityCatalogClient(base_url="http://localhost:8080")
    policy = client.tag_policy("classification").get()
    print(f"Got: {policy.tag_key}")


# --8<-- [end:get_tag_policy]


# --8<-- [start:update_tag_policy]
def update_tag_policy_example() -> None:
    from unitycatalog_client import TagPolicy, UnityCatalogClient

    client = UnityCatalogClient(base_url="http://localhost:8080")
    policy = client.tag_policy("classification").update(
        TagPolicy(tag_key="classification", description="Updated description")
    )
    print(f"Updated: {policy.tag_key}")


# --8<-- [end:update_tag_policy]


# --8<-- [start:delete_tag_policy]
def delete_tag_policy_example() -> None:
    from unitycatalog_client import UnityCatalogClient

    client = UnityCatalogClient(base_url="http://localhost:8080")
    client.tag_policy("classification").delete()
    print("Deleted tag policy")


# --8<-- [end:delete_tag_policy]

File: examples/rust/src/tag_policies.rs
use unitycatalog_client::UnityCatalogClient;
use unitycatalog_common::models::tags::v1::{TagPolicy, Value};

// docs-list_tag_policies-start
pub async fn list_tag_policies_example(base_url: url::Url) {
    let client = UnityCatalogClient::new_unauthenticated(base_url);
    let response = client.list_tag_policies().await.unwrap();
    for policy in response.tag_policies {
        println!("{}", policy.tag_key);
    }
}
// docs-list_tag_policies-end

// docs-create_tag_policy-start
pub async fn create_tag_policy_example(base_url: url::Url) {
    let client = UnityCatalogClient::new_unauthenticated(base_url);
    let policy = client
        .create_tag_policy(TagPolicy {
            tag_key: "classification".to_string(),
            description: Some("Data sensitivity level".to_string()),
            values: vec![
                Value {
                    name: "public".to_string(),
                    ..Default::default()
                },
                Value {
                    name: "restricted".to_string(),
                    ..Default::default()
                },
            ],
            ..Default::default()
        })
        .await
        .unwrap();
    println!("Created: {}", policy.tag_key);
}
// docs-create_tag_policy-end

// docs-get_tag_policy-start
pub async fn get_tag_policy_example(base_url: url::Url) {
    let client = UnityCatalogClient::new_unauthenticated(base_url);
    let policy = client.tag_policy("classification").get().await.unwrap();
    println!("Got: {}", policy.tag_key);
}
// docs-get_tag_policy-end

// docs-update_tag_policy-start
pub async fn update_tag_policy_example(base_url: url::Url) {
    let client = UnityCatalogClient::new_unauthenticated(base_url);
    let policy = client
        .tag_policy("classification")
        .update(TagPolicy {
            tag_key: "classification".to_string(),
            description: Some("Updated description".to_string()),
            ..Default::default()
        })
        .await
        .unwrap();
    println!("Updated: {}", policy.tag_key);
}
// docs-update_tag_policy-end

// docs-delete_tag_policy-start
pub async fn delete_tag_policy_example(base_url: url::Url) {
    let client = UnityCatalogClient::new_unauthenticated(base_url);
    client.tag_policy("classification").delete().await.unwrap();
    println!("Deleted tag policy");
}
// docs-delete_tag_policy-end
